Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -399,31 +395,60 @@ void FunctionDefinitionWriter::visit(AST::UnaryExpression& unary)
     switch (unary.operation()) {
     case AST::UnaryOperation::Negate:
         m_stringBuilder.append("-");
+        break;
+
+    case AST::UnaryOperation::AddressOf:
+    case AST::UnaryOperation::Complement:
+    case AST::UnaryOperation::Dereference:
+    case AST::UnaryOperation::Not:
+        // FIXME: Implement these
+        break;
     }
     visit(unary.expression());
 }
 
 void FunctionDefinitionWriter::visit(AST::BinaryExpression& binary)
 {
-    visit(binary.lhs());
+    visit(binary.leftExpression());
     switch (binary.operation()) {
     case AST::BinaryOperation::Add:
         m_stringBuilder.append(" + ");
         break;
+    case AST::BinaryOperation::Subtract:
+        m_stringBuilder.append(" - ");
+        break;
     case AST::BinaryOperation::Multiply:
         m_stringBuilder.append(" * ");
         break;
+
+    case AST::BinaryOperation::Divide:
+    case AST::BinaryOperation::Modulo:
+    case AST::BinaryOperation::And:
+    case AST::BinaryOperation::Or:
+    case AST::BinaryOperation::Xor:
+    case AST::BinaryOperation::LeftShift:
+    case AST::BinaryOperation::RightShift:
+    case AST::BinaryOperation::Equal:
+    case AST::BinaryOperation::NotEqual:
+    case AST::BinaryOperation::GreaterThan:
+    case AST::BinaryOperation::GreaterEqual:
+    case AST::BinaryOperation::LessThan:
+    case AST::BinaryOperation::LessEqual:
+    case AST::BinaryOperation::ShortCircuitAnd:
+    case AST::BinaryOperation::ShortCircuitOr:
+        // FIXME: Implement these
```

Comment:
nit: I usually like to `ASSERT_NOT_REACHED` so know exactly what's missing when we get to them